compiler/damlc/daml-ide-core/src/Development/IDE/Core/Rules/Daml.hs

```diff
@@ -509,6 +517,64 @@ encodeModuleRule =
         let (hash, bs) = SS.encodeModule lfVersion m
         return ([], Just (mconcat $ hash : map fst encodedDeps, bs))
 
+-- hlint
+
+-- We need to provide hlint the path to the directory of a file
+-- containing a base configuration file called 'hlint.yaml'.
+-- We test with one version of 'hlint.yaml' and deploy with another
+-- (at the current time, there are good reasons for this). The code
+-- below (which relies on the behavior of `package_app` deploying to
+-- '~/.daml/sdk/x.x.x/damlc/resources/daml-ide-core/data') selects one
+-- or the other depending on context.
+getHlintDataDir :: IO FilePath
+getHlintDataDir = do
```

Comment:
We probably want to do this once on startup similar to how we handle the default package db instead of doing it everytime we run HLint.

Reply:
Ok. So two rules? One for the data dir the second for the settings?

Reply:
I probably wouldn’t put this into a rule. Instead add a field for the HLint data dir in the options and set it somewhere on startup, e.g., in `mkOptions`. Seehttps://github.com/digital-asset/daml/blob/13253d82d1d36d3e9fe66fc6359474f58080a428/compiler/damlc/daml-opts/daml-opts-types/DA/Daml/Options/Types.hs#L88 for how we handle this in the package db.

Reply:
Thanks. Will get on this.